processing/src/main/java/org/apache/druid/segment/filter/Filters.java

```diff
@@ -433,10 +434,15 @@ public static Filter convertToCNFFromQueryContext(Query query, @Nullable Filter
       return null;
     }
     boolean useCNF = query.getContextBoolean(QueryContexts.USE_FILTER_CNF_KEY, QueryContexts.DEFAULT_USE_FILTER_CNF);
-    return useCNF ? Filters.toCnf(filter) : filter;
+    try {
+      return useCNF ? Filters.toCnf(filter) : filter;
+    }
+    catch (CNFFilterExplosionException cnfFilterExplosionException) {
+      return filter; // cannot convert to CNF, return the filter as is
```

Comment:
I would just put a log that we are not converting to cnf due to 'CNFFilterExplosionExceptions` so that people debugging the query know whats happening

Reply:
For clusters where this happens regularly, that can generate a lot of log spew.  You could argue that they should respond to the log spew by not doing the queries, but it would still be a sudden chunk of log spew.  In general, CNF is a potential optimization done by the planner, we don't log other optimization choices that we make, we just make them.  In this case, it's deeming that converting to CNF is more expensive than just leaving it as is, which is yet another optimization decision.  I don't think it needs a log, or if it does, maybe at debug level.

Reply:
I understand where your coming from. In such a case, I would then tend towards not throwing an exception and refactoring the `Filter.toCnf(filter)` to return an appropriate state which helps us decide if the conversionWasSucessfull or not. Maybe use `Optional.of()`so that we avoid throwing exceptions due to performance overheads of exceptions. 
We can take this up as future work. Maybe when we are making this limit configurable.